app/controllers/api/base_controller/normalizer.rb

```diff
@@ -20,37 +17,19 @@ def normalize_hash(type, obj, opts = {})
         end
 
         attrs.each do |k|
-          value =  normalize_direct(type, k, obj.kind_of?(ActiveRecord::Base) ? obj.try(k) : obj[k])
+          value = obj.kind_of?(ActiveRecord::Base) ? normalize_attr(k, obj.try(k)) : normalize_attr(k, obj[k])
           result[k] = value unless value.nil?
         end
         result
       end
 
-      def normalize_virtual(vtype, name, obj, options = {})
-        return normalize_virtual_array(vtype, name, obj, options) if obj.kind_of?(Array) || obj.kind_of?(ActiveRecord::Relation)
-        return normalize_virtual_hash(vtype, obj, options) if obj.respond_to?(:attributes) || obj.respond_to?(:keys)
-        normalize_attr_byname(name, obj)
-      end
-
-      def normalize_virtual_array(vtype, name, obj, options)
-        obj.collect { |item| normalize_virtual(vtype, name, item, options) }
-      end
-
-      def normalize_virtual_hash(vtype, obj, options)
-        Environment.fetch_encrypted_attribute_names(obj.class)
-        attrs = (obj.respond_to?(:attributes) ? obj.attributes.keys : obj.keys)
-        attrs.each_with_object({}) do |k, res|
-          value = normalize_virtual(vtype, k, obj[k], options)
-          res[k] = value unless options[:ignore_nil] && value.nil?
-        end
-      end
-
-      #
-      # Let's normalize the attribute based on its name
-      #
-      def normalize_attr_byname(attr, value)
+      def normalize_attr(attr, value)
         return if value.nil?
-        if Environment.normalized_attributes[:time].key?(attr.to_s)
+        if value.kind_of?(Array) || value.kind_of?(ActiveRecord::Relation)
+          normalize_direct_array(attr, value)
```

Comment:
Was this method so-named to distinguish from `normalize_virtual_array`? If that's gone now, WDYT about calling this 'normalize_array`?